exec/java-exec/src/main/java/org/apache/drill/exec/ops/QueryContext.java

```diff
@@ -97,7 +94,7 @@ public QueryContext(final UserSession session, final DrillbitContext drillbitCon
         plannerSettings.getPlanningMemoryLimit());
     bufferManager = new BufferManagerImpl(this.allocator);
     viewExpansionContext = new ViewExpansionContext(this);
-    schemaTreesToClose = Lists.newArrayList();
+    schemaTreeProvider = new SchemaTreeProvider(drillbitContext);
```

Comment:
yes. It contains the schema trees that are created within the query. It is going to differ for every query depending upon what schemas are added. Also it is a very light weight object.